node.go

```diff
@@ -514,6 +520,20 @@ func (n *Node) switchToCandidate() {
 	n.switchState(CANDIDATE)
 }
 
+// Execute `f` in a separate go routine, but ensures that functions
+// are executed in order.
+func (n *Node) serializeGoRoutine(nextCh *(chan bool), f func()) {
```

Comment:
So we have N Go routines they are just chained together correct? Feels like we should be able to do with 1 Go routine correct?